rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -778,6 +768,13 @@ async fn run_controller(
         tracing::info!("No token / actor_name on disk, starting in signed-out state");
     }
 
+    if !ran_before::get().await? {
+        let win = app
+            .get_window("welcome")
+            .context("Couldn't get handle to Welcome window")?;
+        win.show()?;
+    }
+
```

Comment:
The Welcome screen had to move later so it would be after we try to load the token from disk.